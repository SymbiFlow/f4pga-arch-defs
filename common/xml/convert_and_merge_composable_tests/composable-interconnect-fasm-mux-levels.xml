<?xml version="1.0"?>
<!-- Test the special handling around the fasm_mux metadata when using
     composable interconnects and existing metadata. -->
<xml>
 <pb_type name="parent">
  <input name="i0" />
  <input name="i1" />
  <output name="o" />

  <pb_type name="blocka">
   <input name="i" />
   <output name="a" />
  </pb_type>
  <pb_type name="blockb">
   <input name="i" />
   <output name="b" />
  </pb_type>
  <pb_type name="blockc" num_pb="2">
   <input name="i" />
   <output name="c" />
  </pb_type>
  <pb_type name="blocko" num_pb="2">
   <input name="i" />
   <output name="o" />
  </pb_type>

  <interconnect>
   <mux name="outmux">
    <port type="input" name="a" from="blocka">
     <metadata>
      <meta name="fasm_mux">a1</meta>
     </metadata>
    </port>
    <port type="input" name="b" from="blockb">
     <metadata>
      <meta name="fasm_mux">b1.l1</meta>
     </metadata>
    </port>
    <port type="input" name="c" from="blockc[0]">
     <metadata>
      <meta name="fasm_mux">c0.l1</meta>
     </metadata>
    </port>
    <port type="input" name="c" from="blockc[1]">
     <metadata>
      <meta name="fasm_mux">c1</meta>
     </metadata>
    </port>
    <port type="output" from="blocko" name="i" />
    <metadata>
     <meta name="fasm_name">fasm_name</meta>
    </metadata>
   </mux>
   <direct>
    <port type="input" name="o" from="blocko" />
    <port type="output" name="o" />
   </direct>
  </interconnect>
 </pb_type>
</xml>
