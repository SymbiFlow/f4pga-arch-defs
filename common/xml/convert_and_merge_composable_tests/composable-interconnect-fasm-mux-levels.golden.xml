<?xml version="1.0"?>
<xml>
  <pb_type name="parent">
    <input name="i0"/>
    <input name="i1"/>
    <output name="o"/>
    <pb_type name="blocka">
      <input name="i"/>
      <output name="a"/>
    </pb_type>
    <pb_type name="blockb">
      <input name="i"/>
      <output name="b"/>
    </pb_type>
    <pb_type name="blockc" num_pb="2">
      <input name="i"/>
      <output name="c"/>
    </pb_type>
    <pb_type name="blocko" num_pb="2">
      <input name="i"/>
      <output name="o"/>
    </pb_type>
    <interconnect>
      <mux input="blocka.a blockb.b blockc[0].c blockc[1].c" name="outmux" output="blocko.i">
        <metadata>
          <meta name="fasm_mux">
blocka.a : a1
blockb.b : b1.l1
blockc[0].c : c0.l1
blockc[1].c : c1
</meta>
          <meta name="fasm_name">fasm_name</meta>
        </metadata>
      </mux>
      <direct input="blocko.o" name="parent-o" output="parent.o"/>
    </interconnect>
  </pb_type>
</xml>
